tools/extras/install_tcmalloc.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+#
+# Copyright 2021 Hang Lyu
+#
+# This script attempts to install tcmalloc and libunwind.
+# The tcmalloc provides the more efficient way to malloc so that it can speed
+# up the code, especially for the decoding code which contains massive memory
+# allocation operations.
+#
+# At the same time, the tcmalloc also provides some profilers which can help
+# the user to analysis the performance of the code. However, there may have 
+# some deadlock problems when you use the profilers with default build-in glibc.
+# So the libunwind is recommanded to be installed. When the deadlock problems
+# happen, the user can try to link with the library libunwind. But there may
+# still be some crash on x64 platform. You can skip the installation about
+# libunwind if you don't need it. (Link with "-lunwind" after your installation
+# if you want to include it.)
+#
+# Depending on different platforms which are used by different users, the users
+# also can try differnet malloc libraries such as tbbmalloc and so on. From our
+# test, the tcmalloc is most efficient on our platform.
+
+
+# Make sure we are in the tools/ directory.
+if [ $(basename $PWD) == extras ]; then
+  cd ..
+fi
+
+! [ $(basename $PWD) == tools ] && \
+  echo "You must call this script from the tools/ directory" && exit 1;
+
+# prepare libunwind
+echo "****() Installing libunwind"
```

Comment:
* libtcmalloc_minimal does not depend on libunwind, see comments below. We can shed this dependency.
* libunwind comes with most distros and can be installed with apt (libunwind-dev) or yum (libunwid-devel), unless the version 1.5-rc is strictly required. Also, clang comes with its own version; I do not know how compatible they are. I'd left the choice to the user.